elasticdl/python/master/evaluation_service.py

```diff
@@ -13,12 +12,49 @@
 class _EvaluationJob(object):
     """Representation of an evaluation job"""
 
-    def __init__(self, model_version, total_tasks=-1):
+    def __init__(self, metrics_dict, model_version, total_tasks=-1):
+        """
+        Args:
+            metrics_dict: A python dictionary. If model has only one output,
+                `metrics_dict` is a dictionary of `{metric_name: metric}`,
+                i.e. `{"acc": tf.keras.metrics.Accuracy()}`.
+                If model has multiple outputs, `metric_dict` is a dictionary of
+                `{output_name: {metric_name: metric}}`,
+                i.e. `{
+                    "output_a": {"acc": tf.keras.metrics.Accuracy()},
+                    "output_b": {"auc": tf.keras.metrics.AUC()},
+                }`. Note that for model with multiple outputs, each metric
+                only uses one output.
+            model_version: The version of the model to be evaluated.
+            total_tasks: The number of evaluation tasks.
+        """
+
         self.model_version = model_version
         self._total_tasks = total_tasks
         self._completed_tasks = 0
-        self._completed_minibatches = 0
-        self._evaluation_metrics = defaultdict()
+        self._init_metrics_dict(metrics_dict)
+
+    def _init_metrics_dict(self, metrics_dict):
+        if not metrics_dict:
+            raise ValueError(
+                "Doing evaluation with empty evaluation metrics dict."
+            )
+        first_metrics = list(metrics_dict.values())[0]
```

Comment:
Yes, it is a little weird here. This is because I want to support two kinds of user interface, as illustrated in the docstring of `_EvaluationJob.__init__`. 

Function `__init__` unifies these two cases into a consistent structure. 
Function `report_evaluation_metrics` calculates metrics through this structure. 
Function `get_evaluation_summary` returns the calculated metrics and logger will print the returned metrics. `get_evaluation_summury` should return the metrics in the similar form as the user interface.